Run starting. Expected test count is: 6
GetSaleTestSuite:
GetSale
  BaseTest
    validation at different sale stages
    + Sending Create Sale POST Request and validating Create Sale Response 
    + Test Request 
    + Sending Request: 
	RequestDetails[Method:[POST]URL:[http://localhost:81/Xpidia/supplierBranches/woi/sales]
	RequestHeaders:["x-request-identifier=69ec98a5-72d8-4301-9211-0d50e081a44e"; "x-request-authentication=xIoLt8rNatrP0IbNaZl+JOkLK6r78Eq5ESPe6lU5yEc="; "Accept=application/vnd.localexpert.v2.1+json,application/vnd.localexpert.v2.0+json;q=0.9,application/vnd.localexpert.v1.1+json;q=0.8,application/json;q=0.7"; "Content-Type=application/vnd.localexpert.v2.1+json; charset=UTF-8"]
	RequestBody:[{"referenceId":"a3c50b95-e751-4f18-b881-dccde6a6b1de","partnerActivityId":"291957W3","partnerOfferId":"391472W3025","localDate":"2017-03-21","ticketTypes":[{"partnerTicketTypeId":"adult","count":2,"travelerCount":2,"voucherCount":2}],"guests":[{"firstName":"NameFirst","lastName":"NameLast"}],"holdDurationSeconds":5,"additionalCriteria":[]}]] 
    - Create Sale Failure *** FAILED ***
      Create Sale failed therefore a Get Sale cannot be tested. Exception message: Create Sale failed due to: Exception thrown during POST request. Exception message: HTTP Request timed out after 5 seconds. (GetSaleTestSuite.scala:66)
  PerGroupVBP
  + Sending Create Sale POST Request and validating Create Sale Response 
  + Sending Create Sale POST Request and validating Create Sale Response 
  + Test Request 
  + Sending Request: 
	RequestDetails[Method:[POST]URL:[http://localhost:81/Xpidia/supplierBranches/4246/sales]
	RequestHeaders:["x-request-identifier=d11181d5-e7f3-4d2f-bae6-4158852d0021"; "x-request-authentication=qS/dPC36zl02alYwToznO2C/C8cOl45T60VwWarOqvQ="; "Accept=application/vnd.localexpert.v2.1+json,application/vnd.localexpert.v2.0+json;q=0.9,application/vnd.localexpert.v1.1+json;q=0.8,application/json;q=0.7"; "Content-Type=application/vnd.localexpert.v2.1+json; charset=UTF-8"]
	RequestBody:[{"referenceId":"681da8f5-5d96-42cf-9e3d-9d12c6660f50","partnerActivityId":"83","partnerOfferId":"t540-60","localDate":"2020-01-01","ticketTypes":[{"partnerTicketTypeId":"qty","count":1,"travelerCount":5,"voucherCount":1}],"guests":[{"firstName":"NameFirst","lastName":"NameLast"}],"holdDurationSeconds":5,"additionalCriteria":[]}]] 
  + Received Response: 
	ResponseDetails[ResponseHeaders:["Date=Wed, 26 Sep 2018 03:36:02 GMT"; "Server=Apache/2.4.27 (Win64) OpenSSL/1.0.2l PHP/7.1.12"; "X-Powered-By=PHP/7.1.12"; "Content-Length=264"; "Keep-Alive=timeout=5, max=100"; "Connection=Keep-Alive"; "Content-Type=application/vnd.localexpert.v2.1+json"] 
	ResponseContentType:[application/vnd.localexpert.v2.1+json]
	ResponseBody:[{"responseHeader":{"requestIdentifier":"d11181d5-e7f3-4d2f-bae6-4158852d0021","processingMilliseconds":100,"errorType":"PartnerActivityIdUnrecognized","errorMessage":"The Activity ID specified could not be found in the system or belongs to an inactive Activity."}}]] 
  - Commit Sale Failure *** FAILED ***
    Commit Sale failed therefore a Get Sale cannot be tested. Exception message: Commit Sale failed due to: Exception was thrown during createSale. Exception message: Create Sale failed due to: Response Body from Create Sale was converted into an Error Response when a valid response was expected. errorType: PartnerActivityIdUnrecognized errorMessage: The Activity ID specified could not be found in the system or belongs to an inactive Activity. (GetSaleTestSuite.scala:107)
  SaleNotFound
  + Test Request 
  + Sending Request: 
	RequestDetails[Method:[GET]URL:[http://localhost:81/Xpidia/supplierBranches/woi/sales/6a846113-680a-4c78-bc98-ab0abc7eff93]
	RequestHeaders:["x-request-identifier=1a2afb53-914d-476e-8485-0c3613db70dc"; "x-request-authentication=BA7jQSa08eH4U7F7zOYPu2Hpgwif6vQdc0o8iyTvHMg="; "Accept=application/vnd.localexpert.v2.1+json,application/vnd.localexpert.v2.0+json;q=0.9,application/vnd.localexpert.v1.1+json;q=0.8,application/json;q=0.7"; "Content-Type=application/vnd.localexpert.v2.1+json; charset=UTF-8"]
	RequestBody:[]] 
  + Received Response: 
	ResponseDetails[ResponseHeaders:["Date=Wed, 26 Sep 2018 03:36:04 GMT"; "Server=Apache/2.4.27 (Win64) OpenSSL/1.0.2l PHP/7.1.12"; "X-Powered-By=PHP/7.1.12"; "Content-Length=460"; "Keep-Alive=timeout=5, max=99"; "Connection=Keep-Alive"; "Content-Type=application/vnd.localexpert.v2.1+json"] 
	ResponseContentType:[application/vnd.localexpert.v2.1+json]
	ResponseBody:[{"responseHeader":{"requestIdentifier":"1a2afb53-914d-476e-8485-0c3613db70dc","processingMilliseconds":100},"partnerSupplierBranchId":"woi","partnerSaleId":"6a846113-680a-4c78-bc98-ab0abc7eff93","partnerSaleStatus":"OnHold","partnerBarcodeSymbology":"QRCode","partnerSaleBarcode":"","utcSaleRedemptionDateTime":100,"partnerTickets":{"ticketId":"","partnerTicketId":"","partnerTicketStatus":"OnHold","partnerTicketBarcode":"","utcTicketRedemptionDateTime":100}}]] 
  + Test Response Mapping 
  - Http StatusCode should be 200
  - Response time should be within the maximum allowed response time. GetSale calls are required to be within 1000 milliseconds
  - Response Body should map to an Error Type Response Model *** FAILED ***
    Response Body could not be mapped to a valid Error Response Type. Keys may be incorrect. No constructor for type Manifest[GetSaleResponseBasic], JObject(List((responseHeader,JObject(List((requestIdentifier,JString(1a2afb53-914d-476e-8485-0c3613db70dc)), (processingMilliseconds,JInt(100))))), (partnerSupplierBranchId,JString(woi)), (partnerSaleId,JString(6a846113-680a-4c78-bc98-ab0abc7eff93)), (partnerSaleStatus,JString(OnHold)), (partnerBarcodeSymbology,JString(QRCode)), (partnerSaleBarcode,JString()), (utcSaleRedemptionDateTime,JInt(100)), (partnerTickets,JObject(List((ticketId,JString()), (partnerTicketId,JString()), (partnerTicketStatus,JString(OnHold)), (partnerTicketBarcode,JString()), (utcTicketRedemptionDateTime,JInt(100))))))) (SharedHttpTests.scala:182)
    Response Body Tests
    - Json Mapping Failure *** FAILED ***
      Response Body tests were skipped due to a Json mapping exception. No usable value for responseHeader
      No usable value for errorType
      Did not find value which can be converted into java.lang.String (SharedHttpTests.scala:89)
Run completed in 12 seconds, 823 milliseconds.
Total number of tests run: 6
Suites: completed 1, aborted 0
Tests: succeeded 2, failed 4, canceled 0, ignored 0, pending 0
*** 4 TESTS FAILED ***
